### PR TITLE
Fix flaky SyntheticCryptoTransferApprovalsMigrationTest

### LINKED ISSUE #7188: Flaky SyntheticCryptoTransferApprovalsMigrationTest
### Description

`SyntheticCryptoTransferApprovalsMigrationTest` can sometimes fail in CI: 

```
java.lang.AssertionError: |  
-- | --
  | Expecting actual: |  
  | [CryptoTransfer(amount=-10, consensusTimestamp=1680284879342164980, entityId=465143, errata=null, isApproval=false, payerAccountId=0.0.2119901), |  
  | CryptoTransfer(amount=-10, consensusTimestamp=1680284879342163978, entityId=465177, errata=null, isApproval=false, payerAccountId=0.0.2119901), |  
  | CryptoTransfer(amount=-10, consensusTimestamp=1680284879342064922, entityId=465143, errata=null, isApproval=false, payerAccountId=0.0.2119901), |  
  | CryptoTransfer(amount=-10, consensusTimestamp=1680284879342164981, entityId=465143, errata=null, isApproval=false, payerAccountId=0.0.2119901), |  
  | CryptoTransfer(amount=-10, consensusTimestamp=1680284879342164983, entityId=465143, errata=null, isApproval=true, payerAccountId=0.0.2119901), |  
  | CryptoTransfer(amount=-10, consensusTimestamp=1680284879342164984, entityId=465153, errata=null, isApproval=false, payerAccountId=0.0.2119901), |  
  | CryptoTransfer(amount=-10, consensusTimestamp=1686243920981874003, entityId=465143, errata=null, isApproval=false, payerAccountId=0.0.2119901), |  
  | CryptoTransfer(amount=-10, consensusTimestamp=1680284879342164979, entityId=465138, errata=null, isApproval=false, payerAccountId=0.0.2119900), |  
  | CryptoTransfer(amount=-10, consensusTimestamp=1680284879342163977, entityId=465160, errata=null, isApproval=false, payerAccountId=0.0.2119900), |  
  | CryptoTransfer(amount=-10, consensusTimestamp=1680284879342164982, entityId=465148, errata=null, isApproval=false, payerAccountId=0.0.465148), |  
  | CryptoTransfer(amount=10, consensusTimestamp=1699472277141733569, entityId=465250, errata=null, isApproval=false, payerAccountId=0.0.465251)] |  
  | to contain exactly in any order: |  
  | [CryptoTransfer(amount=-10, consensusTimestamp=1680284879342164980, entityId=465143, errata=null, isApproval=true, payerAccountId=0.0.2119901), |  
  | CryptoTransfer(amount=-10, consensusTimestamp=1680284879342163978, entityId=465177, errata=null, isApproval=true, payerAccountId=0.0.2119901), |  
  | CryptoTransfer(amount=-10, consensusTimestamp=1680284879342164984, entityId=465153, errata=null, isApproval=true, payerAccountId=0.0.2119901), |  
  | CryptoTransfer(amount=-10, consensusTimestamp=1680284879342164979, entityId=465138, errata=null, isApproval=false, payerAccountId=0.0.2119900), |  
  | CryptoTransfer(amount=-10, consensusTimestamp=1680284879342163977, entityId=465160, errata=null, isApproval=false, payerAccountId=0.0.2119900), |  
  | CryptoTransfer(amount=-10, consensusTimestamp=1680284879342064922, entityId=465143, errata=null, isApproval=false, payerAccountId=0.0.2119901), |  
  | CryptoTransfer(amount=-10, consensusTimestamp=1680284879342164981, entityId=465143, errata=null, isApproval=false, payerAccountId=0.0.2119901), |  
  | CryptoTransfer(amount=-10, consensusTimestamp=1680284879342164982, entityId=465148, errata=null, isApproval=false, payerAccountId=0.0.465148), |  
  | CryptoTransfer(amount=10, consensusTimestamp=1699472277141733569, entityId=465250, errata=null, isApproval=false, payerAccountId=0.0.465251), |  
  | CryptoTransfer(amount=-10, consensusTimestamp=1680284879342164983, entityId=465143, errata=null, isApproval=true, payerAccountId=0.0.2119901), |  
  | CryptoTransfer(amount=-10, consensusTimestamp=1686243920981874003, entityId=465143, errata=null, isApproval=false, payerAccountId=0.0.2119901)] |  
  | elements not found: |  
  | [CryptoTransfer(amount=-10, consensusTimestamp=1680284879342164980, entityId=465143, errata=null, isApproval=true, payerAccountId=0.0.2119901), |  
  | CryptoTransfer(amount=-10, consensusTimestamp=1680284879342163978, entityId=465177, errata=null, isApproval=true, payerAccountId=0.0.2119901), |  
  | CryptoTransfer(amount=-10, consensusTimestamp=1680284879342164984, entityId=465153, errata=null, isApproval=true, payerAccountId=0.0.2119901)] |  
  | and elements not expected: |  
  | [CryptoTransfer(amount=-10, consensusTimestamp=1680284879342164980, entityId=465143, errata=null, isApproval=false, payerAccountId=0.0.2119901), |  
  | CryptoTransfer(amount=-10, consensusTimestamp=1680284879342163978, entityId=465177, errata=null, isApproval=false, payerAccountId=0.0.2119901), |  
  | CryptoTransfer(amount=-10, consensusTimestamp=1680284879342164984, entityId=465153, errata=null, isApproval=false, payerAccountId=0.0.2119901)] |  
  |   |  


at com.hedera.mirror.importer.migration.SyntheticCryptoTransferApprovalsMigrationTest.assertTransfers(SyntheticCryptoTransferApprovalsMigrationTest.java:311) |  
at com.hedera.mirror.importer.migration.SyntheticCryptoTransferApprovalsMigrationTest.onEnd(SyntheticCryptoTransferApprovalsMigrationTest.java:238)


```

### Steps to reproduce

1. Run CI

### Additional context

_No response_

### Hedera network

other

### Version

main

### Operating system

None

### PR DESCRIPTION
**Pull request by edwin-greene**
**Description**:
Update test to consistently pass CI

**Related issue(s)**:

Fixes #7188 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/migration/SyntheticCryptoTransferApprovalsMigrationTest.java [hunk 1]
```diff
@@ -90,16 +89,16 @@ class SyntheticCryptoTransferApprovalsMigrationTest extends IntegrationTest {
...
     @BeforeEach
+    @SneakyThrows
     void setup() {
         mirrorProperties.setNetwork(MAINNET);
+        var activeField = migration.getClass().getDeclaredField("executed");
+        activeField.setAccessible(true);
+        activeField.set(migration, new AtomicBoolean(false));
+        var completeField = migration.getClass().getSuperclass().getDeclaredField("complete");
+        completeField.setAccessible(true);
+        completeField.set(migration, new AtomicBoolean(false));
```
Comment (steven-sheehy) [inline hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/migration/SyntheticCryptoTransferApprovalsMigrationTest.java:101]: We shouldn't use reflection. If we need access expose them as package private with `@VisibleForTesting`. We should hide the AtomicBoolean nature of the field and just take in boolean.

File: hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/migration/SyntheticCryptoTransferApprovalsMigrationTest.java [hunk 2]
```diff
@@ -817,19 +817,4 @@ private EntityHistory saveHistory(Entity entity) {
...
-    private void waitForCompletion() {
-        await().atMost(Duration.ofSeconds(5))
-                .pollDelay(Duration.ofMillis(100))
-                .pollInterval(Duration.ofMillis(100))
-                .untilAsserted(() -> assertThat(isMigrationCompleted()).isTrue());
-    }
-
-    private boolean isMigrationCompleted() {
-        var actual = jdbcTemplate.queryForObject(
-                "select (select checksum from flyway_schema_history where script = ?)",
```
Comment (steven-sheehy) [inline hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/migration/SyntheticCryptoTransferApprovalsMigrationTest.java:830]: Not clear to me how this wouldn't work unless there were multiple installed_rank where script= SyntheticCryptoTransferApprovalMigration. But perhaps in memory approach is more efficient anyway.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/migration/SyntheticCryptoTransferApprovalsMigrationTest.java (hunk 1 above)
Comment (steven-sheehy) [reply hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/migration/SyntheticCryptoTransferApprovalsMigrationTest.java]: I meant as setters. That's why I mentioned take in a boolean. Should not expose fields directly.

Comment (edwin-greene) [reply hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/migration/SyntheticCryptoTransferApprovalsMigrationTest.java]: Thanks! I have updated it now.

**Review by steven-sheehy — APPROVED** [review]
LGTM

File: hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/migration/SyntheticCryptoTransferApprovalsMigrationTest.java (hunk 2 above)
Comment (xin-hedera) [reply hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/migration/SyntheticCryptoTransferApprovalsMigrationTest.java:830]: very likely. I did fix the query in my balance partitioning and dedup PR

**Review by xin-hedera — APPROVED** [review]
LGTM

**Pull request merged**